ci-operator/step-registry/ibmcloud/provision/vpc/endpoints-gateway/ibmcloud-provision-vpc-endpoints-gateway-commands.sh

```diff
@@ -36,10 +36,23 @@ function createEndpointGateway() {
     echo "Command: $cmd"
     eval "$cmd" &> "${log}"; ret=$?
     cat "${log}"
-    if [[ "$ret" != "0" ]] && grep -q "endpoint gateway already exists for this service" "${log}"; then
-        echo "The endpoint gateway already exists for this service, ignore the error..."
-        return 0
-    fi
+    if [[ "$ret" != "0" ]]; then
+        if grep -q "endpoint gateway already exists for this service" "${log}"; then
+            echo "The endpoint gateway already exists for this service, ignoring the error..."
+            return 0
+        fi
+        
+        echo "Retrying..."
+        echo "" &> "${log}"
+        if ! waitingVPCAvaliable ${vpcID} || ! eval "$cmd" &> "${log}"; then
```

Comment:
I do not think this is a good place to reference `waitingVPCAvaliable` inside `createEndpointGateway`

Reply:
Jenkins has the same problem https://jenkins-csb-openshift-qe-mastern.dno.corp.redhat.com/job/ocp-common/job/Flexy-install/352259/console 

checked https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/67548/rehearse-67548-periodic-ci-openshift-verification-tests-main-installer-rehearse-4.20-installer-rehearse-ibmcloud-dis/1950056538215288832 , when endpoint-gateway-create, the VPC status is not "available" 
Added waitingVPCAvaliable function can resolve this issue.

And I am talking with IBM support about whether this is a temporary issue.

Reply:
Then I think you should add `waitingVPCAvaliable` in `ibmcloud-provision-vpc` step

Reply:
> when endpoint-gateway-create, the VPC status is not "available"

Where can I see that ? I only saw:
```
Current vpc status:
FAILED
'vpc' is not a registered command. Check your list of installed plug-ins. See 'ibmcloud help'.
```

Reply:
@jianlinliu changed to based on the oc version, create  "ResourceController" or "ResourceManager", to avoid the error "The endpoint gateway already exists for this service, ignoring the error..." with the same target URL.